tox.ini

```diff
@@ -46,11 +46,14 @@ commands =
 setenv =
     SKIP = mypy
 whitelist_externals =
+    bash
     mypy
     pre-commit
 commands =
     mypy .  # run separately because of potential caching problems
     pre-commit run {posargs} -a
+    - bash -ec "cspell $(git ls-files)"
+    - bash -ec "markdownlint ."
```

Comment:
How do we guarantee that cpsell and markdownlint are installed?

Reply:
Non-zero exit codes of commands starting with `-` are ignored, see
https://tox.readthedocs.io/en/latest/example/basic.html#ignoring-exit-code

Note, however, that this also means `tox -e sty` does not fail if `cspell` or `markdownlint` do emit errors, so you would have to see this in the terminal. This is also the reason why both `cspell` and `markdownlint` are still called separately in GitHub Actions.

I've also not added `pyright`, because it doesn't find the imports when executed through `tox`.

Reply:
i see. thx for the clarification